docs/DML-Operations-on-Carbon.md

```diff
@@ -1,197 +1,308 @@
+
+  ```ruby
+  DELETE SEGMENTS FROM TABLE [db_name.]table_name WHERE STARTTIME BEFORE [DATE_VALUE];
+  ```
+
+### Parameter Description
+
+| Parameter | Description | Optional |
+| ------------- | -----| ------ |
+| DATE_VALUE | Valid segement load start time value. All the segments before this specified date will be deleted. | NO |
+| db_name | Database name, if it is not specified then it uses current database. | YES |
+| table_name | The name of the table in provided database.| NO |
+
+**Example:**
+
+  ```ruby
+  DELETE SEGMENTS FROM TABLE CarbonDatabase.CarbonTable WHERE STARTTIME BEFORE '2017-06-01 12:05:06';  
+  ```
+
+***
+
+# UPDATE CARBON TABLE
+This command updates the carbon table based on the column expression and optional filter conditions.
+ 
+The client node where the UPDATE command is executed should be part of the cluster.
+### Syntax
+Syntax1:
+  ```ruby
+  UPDATE <CARBON TABLE>
+  SET (column_name1, column_name2, ... column_name n) =  (column1_expression , column2_expression , column3_expression . .. column n_expression )
+  [ WHERE { <filter_condition> } ]; 
+  ```
+Syntax2:
+  ```ruby
+  UPDATE <CARBON TABLE>
+  SET (column_name1, column_name2,) =  (select sourceColumn1, sourceColumn2 from sourceTable [ WHERE { <filter_condition> } ] )
+  [ WHERE { <filter_condition> } ];
+  ```
+
+### Parameter Description
+
+| Parameter | Description |
+| ------------- | -----|
+| CARBON TABLE | The name of the Carbon table in which you want to perform the update operation. |
+| column_name | The destination columns to be updated. |
+| sourceColumn | The source table column values to be updated in destination table. |
+| sourceTable | The table from which the records are updated into destination Carbon table. |
+ 
+### Usage Guidelines
+Following are the conditions to use UPDATE:
+- The update command fails if multiple input rows in source table are matched with single row in destination table.
+- If the source table generates empty records, the update operation will complete without updating the table.
+- If a source table row does not correspond to any of the existing rows in a destination table, the update operation will complete without updating the table.
+- In sub-query, if the source table and the target table are same, then the update operation fails.
+- If the sub-query used in UPDATE statement contains aggregate method or group by query, then the UPDATE operation fails.
+  For example, update t_carbn01 a set (a.item_type_code, a.profit) = ( select b.item_type_cd, sum(b.profit) from t_carbn01b b where item_type_cd =2 group by item_type_code);
+  Here, aggregate function sum(b.profit) and group by clause are used in the sub-query, hence the UPDATE operation fails.
+
+**Example1:**
+  ```ruby
+  update carbonTable1 d set (d.column3,d.column5 ) = (select s.c33 ,s.c55 from sourceTable1 s where d.column1 = s.c11) where d.column1 = 'china' exists( select * from table3 o where o.c2 > 1);
+  ```
+
+**Example2:**

```

Comment:
Any reason that we pick these 5 examples? If you want to cover all syntax, can you briefly describe for each example?